### PR TITLE
[pt2] add Python meta for `triangular_solve`

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #106682

### CODE REVIEW

**Comment by nkaretnikov** [comment]
CPP meta exists, but tests were failing. Added a Python meta.

File: torch/_meta_registrations.py [hunk 1]
```diff
@@ -1136,6 +1136,54 @@ def linalg_solve_triangular_meta(
...
+            size=A_broadcast_size,
+            stride=make_contiguous_strides_for(A_broadcast_size, row_major=False),
+            dtype=A.dtype,
+            device=A.device,
+        )
+    elif A.layout == torch.sparse_csr or A.layout == torch.sparse_bsr:
+        solution = torch.empty_like(self)
+        cloned_coefficient = self.new_empty([0])
+    else:
+        torch._check(False, "triangular_solve: Got an unexpected layout.")
```
Comment (nkaretnikov) [inline torch/_meta_registrations.py:1183]: Will fix: forgot lambda here.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**